README.md

```diff
@@ -691,6 +692,68 @@ declare module '@pigment-css/react/theme' {
 }
 ```
 
+## sx prop
+
+A special `sx` prop lets you apply styles directly to an element. When `sx` prop is specified on an element, Pigment CSS will replace it with `className` and `style` props at build time.
+
+The `sx` prop works on any element, including HTML elements and 3rd-party custom components as long as it is JSX.
+
+### Usage
+
+```js
+<div sx={{ display: 'flex', flexDirection: 'column' }}>
+```
+
+For a React component like the example below, it must pass the `className` and `style` props to the underlying DOM element, otherwise the styles won't be applied.
+
+```js
+// /path/to/AnyComponent.js
+const AnyComponent = (props) => {
+  return <div {...props} />;
+};
+
+// /path/to/App.js
+<AnyComponent sx={{ fontSize: 12, color: 'red' }} />;
+```
+
+The value provided to `sx` prop can be one of the following:
+
+- a plain style object (recommended for best performance)
+- a callback function that receives the [theme object](#theming) then return a plain style object:
+
+  ```js
+  <div sx={(theme) => ({ color: theme.colors.primary })} />
+  ```
+
+- an array of plain style objects or callback functions. This is useful for applying conditional styles based on other variables:
+
+  ```js
+  <div
+    sx={[
+      { color: 'red' },
+      selected && { fontWeight: 'bold' },
+      disabled ? (theme) => ({ opacity: theme.state.disabledOpacity }) : { opacity: 1 },
+    ]}
+  />
+  ```
+
+### TypeScript
+
+To use `sx` prop on HTML element, you need to augment the `HTMLAttributes` interface. Add the following code to a file that is included in your tsconfig.json:
+
+```ts
+declare global {
+  namespace React {
+    interface HTMLAttributes<T> {
+      sx?:
+        | React.CSSProperties
+        | ((theme: Theme) => React.CSSProperties)
```

Comment:
We should also add where to import this `Theme` from.

Reply:
Added as a placeholder to imply that it comes from the consumer.